### PR TITLE
CONTRIBUTING.md: add section on Refactoring

### PR DESCRIPTION
**Pull request by WalterBright**
to emphasize what the policy is on refactoring PRs.

Added a couple other minor improvements.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -53,7 +53,9 @@ Before submitting a PR there are some things you can check which will hopefully
 
 - Ensure newly introduced symbols are documented and that updates to existing symbols are reflected in the documentation.
 
-- Add a link to the PR to the Bugzilla entry.
+- Add a link to the PR to the Bugzilla entry (if the dlang-bot failed to do it).
+
+- Confine a PR to addressing one Bugzilla entry, unless multiple entries are different aspects of the same bug.
```
Comment (thewilsonator) [inline CONTRIBUTING.md:58]: Unreleated? If not please also add a note to file a bug report if one does not exist (if this is not covered elsewhere in this document)

Comment (WalterBright) [reply CONTRIBUTING.md:58]: good idea

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -70,6 +75,21 @@ For first-time contributors, we suggest looking for issues categorized as [trivi
...
+program, add enhancements, or fix bugs.
+
+- Refactorings must come with a rationale as to why it makes the code better.
+
+- Break up larger refactorings into a series of self-contained PRs rather than one gigantic one.
+
+- Significant refactorings that include bug fixes and/or enhancements will be rejected with a
+request to submit bug fixes and enhancements as separate PRs. Putting them as separate commits
+is not sufficient. Refactorings will be reviewed on their own merits, not conflated with
+other changes.
```
Comment (Geod24) [inline CONTRIBUTING.md:90]:
```suggestion
- Refactorings must come with a rationale as to why it makes the code better.

- Large refactorings should be done in incremental steps that are easy to review.

- When a refactoring is broken down into multiple PRs, it is acceptable to provide a rationale in the first PR and link to it in subsequent PR.

- Pull requests that do non-trivial refactorings should not include other changes, such as new feature or bug fix.
   While submitting multiple dependent pull requests can be done to help the reviewers judge of the need for a refactoring, any refactoring will be assessed on its own merit.
```

Rationale for the edit:
1) "gigantic" is not very professional;
2) I think we ought to distinguish between "trivial" refactoring (e.g. moving a variable declaration inside a `if` in a single place while fixing the code) and non trivial ones. Trivial refactoring as part of bug fixes / enhancement is common, and forcing people to separate every single line that doesn't directly fix the bug will be a strain on reviewers, contributors and the CI.

**Review by Geod24 — APPROVED** [review]
Thanks for taking the time to do this. Suggested some modification, especially w.r.t. to the "Significant refactoring" part in order to make it more [RFC2119](https://datatracker.ietf.org/doc/html/rfc2119)-like.

**Pull request merged**